packages/datadog-plugin-cucumber/src/index.js

```diff
@@ -0,0 +1,98 @@
+const { relative } = require('path')
+
+const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
+
+const {
+  TEST_TYPE,
+  TEST_NAME,
+  TEST_SUITE,
+  TEST_STATUS,
+  getTestEnvironmentMetadata
+} = require('../../dd-trace/src/plugins/util/test')
+
+function setStatusFromResult (span, result, tag) {
+  if (result.status === 1) {
+    span.setTag(tag, 'pass')
+  } else if (result.status === 2) {
+    span.setTag(tag, 'skip')
+    span.setTag('error.msg', 'skipped')
+  } else if (result.status === 4) {
+    span.setTag(tag, 'skip')
+    span.setTag('error.msg', 'not implemented')
+  } else {
+    span.setTag(tag, 'fail')
+    span.setTag('error.msg', result.message)
+  }
+}
+
+function createWrapRun (tracer, testEnvironmentMetadata, sourceRoot) {
+  return function wrapRun (run) {
+    return function handleRun () {
+      const testName = this.pickle.name
+      const testSuite = relative(sourceRoot, this.pickle.uri)
+
+      const commonSpanTags = {
+        [TEST_TYPE]: 'test',
+        [TEST_NAME]: testName,
+        [TEST_SUITE]: testSuite,
+        [SAMPLING_RULE_DECISION]: 1,
+        ...testEnvironmentMetadata
+      }
+
+      return tracer.trace(
+        'cucumber.test',
+        {
+          type: 'test',
+          resource: testName,
+          tags: commonSpanTags
+        },
+        (span) => {
+          const promise = run.apply(this, arguments)
+          promise.then(() => {
+            setStatusFromResult(span, this.getWorstStepResult(), TEST_STATUS)
+          })
+          return promise
+        }
+      )
+    }
+  }
+}
+
+function createWrapRunStep (tracer) {
+  return function wrapRunStep (runStep) {
+    return function handleRunStep () {
+      const resource = arguments[0].isHook ? 'hook' : arguments[0].pickleStep.text
+      return tracer.trace(
+        'cucumber.step',
+        { type: 'test', resource: resource, tags: { 'cucumber.step': resource } },
```

Comment:
I'm not sure this should be `type: 'test'`, as I think only the test span should have this. I'll ask

Reply:
yeah I'll remove `type: 'test'` from this children steps